packages/requisitions/src/RequestRequisition/DetailView/RequestLineEdit/ItemCharts/StockDistribution.tsx

```diff
@@ -0,0 +1,185 @@
+const MIN_MC_WIDTH_TO_SHOW_TEXT = 5;
+
+const Divider = () => (
+  <Box sx={{ backgroundColor: 'gray.dark', width: '1px', height: '45px' }} />
+);
+
+const ValueBar = ({
+  value,
+  total,
+  label,
+  colour,
+}: {
+  value: number;
+  total: number;
+  label: string;
+  colour: string;
+}) => {
+  if (value === 0) return null;
+
+  const flexBasis = Math.min(Math.round((100 * value) / total), 100);
+
+  return (
+    <>
+      <Tooltip title={`${label}: ${value}`} placement="top">
+        <Box flexBasis={`${flexBasis}%`} flexGrow={1}>
+          <Box sx={{ backgroundColor: colour, height: '20px' }} />
+          <Box style={{ textAlign: 'end', paddingRight: 10, paddingTop: 10 }}>
+            {flexBasis > MIN_FLEX_BASIS_TO_SHOW_LABEL ? (
+              <Typography
+                fontSize={12}
+                style={{ textOverflow: 'ellipsis', height: 20 }}
+              >
+                {label}
+              </Typography>
+            ) : null}
+            {flexBasis > MIN_FLEX_BASIS_TO_SHOW_VALUE ? (
+              <Typography fontSize={12}>{value}</Typography>
+            ) : null}
+          </Box>
+        </Box>
+      </Tooltip>
+      <Divider />
+    </>
+  );
+};
+
+const MonthlyConsumption = ({
+  month,
+  flexBasis,
+  averageMonthlyConsumption,
+  showText,
+}: {
+  month: number;
+  flexBasis: string;
+  averageMonthlyConsumption: number;
+  showText: boolean;
+}) => {
+  const t = useTranslation('common');
+  const text = ` (${month} ${t('label.months', {
+    count: month,
+  })})`;
+  const label = `${averageMonthlyConsumption * month}${showText ? text : ''}`;
+
+  return <MonthlyBar flexBasis={flexBasis} label={label} />;
+};
+
+const MonthlyBar = ({
+  label,
+  left,
+  flexBasis,
+}: {
+  label: string;
+  left?: boolean;
+  flexBasis?: string;
+}) => {
+  const directionStyle = left
+    ? { borderLeftWidth: 1, paddingLeft: 8 }
+    : { paddingLeft: 3, paddingRight: 8, borderRightWidth: 1 };
+  return (
+    <Box
+      sx={{
+        borderWidth: 0,
+        borderBottomWidth: 1,
+        borderColor: 'gray.dark',
+        borderStyle: 'solid',
+        height: '20px',
+      }}
+      style={{ ...directionStyle, textAlign: left ? undefined : 'right' }}
+      flexBasis={flexBasis}
+    >
+      <Typography variant="body1" fontSize={12} style={{ color: 'gray.dark' }}>
+        {label}
+      </Typography>
+    </Box>
+  );
+};
+
+export const StockDistribution: React.FC<StockDistributionProps> = ({
+  itemStats,
+  suggestedQuantity = 0,
+}) => {
+  if (!itemStats) return null;
+
+  const { availableStockOnHand, averageMonthlyConsumption } = itemStats;
+
+  if (averageMonthlyConsumption === 0) return null;
+
+  const { maxMonthsOfStock } = useRequestFields('maxMonthsOfStock');
+  const targetQuantity = maxMonthsOfStock * averageMonthlyConsumption;
+  const t = useTranslation('replenishment');
+
+  if (suggestedQuantity + availableStockOnHand === 0) return null;
+
+  const monthlyConsumptionWidth =
```

Comment:
if the soh is larger than the target then monthly consumption bit is less than 100%